### PR TITLE
Add linkDomain field to ActionCodeSettings

### PR DESCRIPTION
**Pull request by NhienLam**
### Discussion

We are replacing FDL usage in Auth with a new hosting link. This adds a new `linkDomain` field to ActionCodeSettings for developers to set a custom hosting link domain.

### Testing

Tested with unit tests. We will do manual testing when the backend is ready before merging to main branch. 

### API Changes

API proposal approved.

### CODE REVIEW

File: packages/auth/src/api/errors.ts [hunk 1]
```diff
@@ -100,7 +100,8 @@ export const enum ServerError {
   MISSING_RECAPTCHA_VERSION = 'MISSING_RECAPTCHA_VERSION',
   INVALID_RECAPTCHA_VERSION = 'INVALID_RECAPTCHA_VERSION',
   INVALID_REQ_TYPE = 'INVALID_REQ_TYPE',
-  PASSWORD_DOES_NOT_MEET_REQUIREMENTS = 'PASSWORD_DOES_NOT_MEET_REQUIREMENTS'
+  PASSWORD_DOES_NOT_MEET_REQUIREMENTS = 'PASSWORD_DOES_NOT_MEET_REQUIREMENTS',
+  INVALID_HOSTING_LINK_DOMAIN = 'INVALID_HOSTING_LINK_DOMAIN'
```
Comment (Xiaoshouzi-gh) [inline packages/auth/src/api/errors.ts:104]: nit: if possible please add a comma at end so next time the existing line won't show up in the diff.

File: packages/auth/src/core/strategies/action_code_settings.test.ts [hunk 2]
```diff
@@ -60,14 +60,27 @@ describe('core/strategies/action_code_settings', () => {
       _setActionCodeSettingsOnRequest(auth, request, {
         handleCodeInApp: true,
         iOS: {
-          bundleId: 'my-´bundle'
+          bundleId: 'my-bundle'
```
Comment (Xiaoshouzi-gh) [inline packages/auth/src/core/strategies/action_code_settings.test.ts:63]: preferably to create a magic constant for string values that repeated multiple times.
e.g. const TEST_BUNDLE_ID = 'my-bundle';

**Review by Xiaoshouzi-gh — APPROVED** [review]
LGTM

File: packages/auth/src/api/errors.ts (hunk 1 above)
Comment (NhienLam) [reply packages/auth/src/api/errors.ts:104]: I tried, but `yarn run format` automatically removed the comma.

File: packages/auth/src/core/strategies/action_code_settings.test.ts (hunk 2 above)
Comment (NhienLam) [reply packages/auth/src/core/strategies/action_code_settings.test.ts]: Done. Thanks Liubin!

**Pull request merged**